### PR TITLE
Remove linux compilation warnings

### LINKED ISSUE #331: make a pass at removing warnings (clang/g++)
while preserving 0 warnings on VC++

### PR DESCRIPTION
**Pull request by evanlimanto**
Fixes #331.

### CODE REVIEW

File: src/main/main.cpp [hunk 1]
```diff
@@ -329,6 +329,7 @@ catchup(Application::pointer app, uint32_t to, uint32_t count,
             break;
         }
         case LedgerManager::LM_CATCHING_UP_STATE:
+        case LedgerManager::LM_NUM_STATE:
```
Comment (MonsieurNicolas) [inline src/main/main.cpp]: I typically prefer to put those like this, as it makes it obvious that it's unexpected
```
        case LedgerManager::LM_NUM_STATE: 
                abort();
```

**Comment by evanlimanto** [comment]
updated — thanks for the comment!

**Comment by MonsieurNicolas** [comment]
r+ eaa65d261ed0938bbdd3dd74eafad1118eba8fc4

**Comment by evanlimanto** [comment]
CI error was due to formatting — I've repushed the changes.

**Comment by MonsieurNicolas** [comment]
r+ cc6cf936aa29ad7f0eca140aee6b632699013915

**Pull request merged**